x/src/lint/engine/content.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::lint::engine::prelude::*;
+use std::str;
+
+/// Represents a linter that checks file contents.
+pub trait ContentLinter: Linter {
+    /// Pre-run step -- avoids loading the contents if possible.
+    ///
+    /// The default implementation returns `Ok(RunStatus::Success)`; individual lints may configure
+    /// a more restricted set.
+    fn pre_run<'l>(&self, _file_ctx: &FileContext<'l>) -> Result<RunStatus<'l>> {
+        Ok(RunStatus::Executed)
+    }
+
+    /// Executes the lint against the given content context.
+    fn run<'l>(
+        &self,
+        ctx: &ContentContext<'l>,
+        out: &mut LintFormatter<'l, '_>,
+    ) -> Result<RunStatus<'l>>;
+}
+
+#[derive(Debug)]
+pub struct ContentContext<'l> {
+    file_ctx: FileContext<'l>,
+    content: Content,
+}
+
+#[allow(dead_code)]
```

Comment:
Most of it at the moment. I want to extract this into a separate crate at which point it won't matter since it'll be public. Do you think I should do that before landing?

Reply:
Ah got it. I just want to make sure we don't leave it "dead" for too long but since you're actively working on this I'm sure its fine to leave in.